CONTRIBUTING.md

```diff
@@ -4,45 +4,131 @@ We are very open to community contributions and appreciate anything that improve
+trim trailing whitespace.................................................Passed
+check for merge conflicts................................................Passed
+check that scripts with shebangs are executable..........................Passed
+mixed line ending........................................................Passed
+don't commit to branch...................................................Passed
+pretty format json...................................(no files to check)Skipped
+black................................................(no files to check)Skipped
+Update API documentation (slow)..........................................Passed
+Enumerating objects: 14, done.
+Counting objects: 100% (14/14), done.
+Delta compression using up to 16 threads
+Compressing objects: 100% (12/12), done.
+Writing objects: 100% (12/12), 1.06 KiB | 1.06 MiB/s, done.
+Total 12 (delta 8), reused 0 (delta 0), pack-reused 0
+remote: Resolving deltas: 100% (8/8), completed with 2 local objects.
+To github.com:deepset-ai/haystack.git
+   801be454..6fddb35f  my-branch -> my-branch
+```
+
+Note: pre-commit hooks might fail. If that happens to you and you can't understand why, please do the following:
+- Ask for help by opening an issue or reaching out on our Slack channel. We usually give some feedback within a day for most questions.
+- As the last resort, if you are desperate and everything failed, _uninstall the hook_. `pre-commit uninstall -t pre-commit -t pre-push` will remove all checks and let you push. The CI might fail, but at that point we will be able to help.
 
 ## Formatting of Pull Requests
 
-Please give a concise description in the first comment in the PR that includes:
+When you open a pull request, please give a concise description in the first comment in the PR that includes:
 - What is changing?
 - Why?
 - What are limitations?
 - Breaking changes (Example of before vs. after)
 - Link the issue that this relates to
 
-## Working from Github forks
+## CI (Continuous Integration)
 
-Some actions in our CI (code style and documentation updates) will run on your code and occasionally commit back small changes after a push. To be able to do so,
-these actions are configured to run on your fork instead of on the base repository. To allow those actions to run, please don't forget to:
+We use GitHub Action for our Continuous Integration tasks. This means that, as soon as you open a PR, GitHub will start executing some workflows on your code, like automated tests, linting, formatting, api docs generation, etc.
 
-1. Enable actions on your fork with read and write permissions:
+If all goes well, at the bottom of your PR page you should see something like this, where all checks are green.
 
-<p align="center"><img src="https://raw.githubusercontent.com/deepset-ai/haystack/master/docs/img/fork_action_config.png"></p>
+<p align="center"><img src="https://raw.githubusercontent.com/deepset-ai/haystack/master/docs/img/ci-success.png"></p>
```

Comment:
Any reason not to use markdown with a relative path here? e.g. `![Successful CI](docs/img/ci-success.png)`

Reply:
Technically the alignment to the center can't be made with markdown AFAIK. I can remove it, but I will then remove it for all pictures to be consistent

Reply:
I realized there was only another picture and I've also added it a while ago :smile: Let's remove this

Reply:
Ah true, indeed you can't center it. Follow up: can we try keeping the html but using relative paths? (disclaimer: it's very useful for the review 😬)